Code/Framework/AzToolsFramework/Platform/Windows/AzToolsFramework/API/PythonLoader_Windows.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#include <AzToolsFramework/API/PythonLoader.h>
+
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/string/conversions.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzFramework/IO/LocalFileIO.h>
+
+#include <shlobj.h>
+#include <tchar.h>
+
+namespace AzToolsFramework::EmbeddedPython
+{
+    void PythonLoader::LoadRequiredModules()
+    {
+        // No required modules need explicit loading on this platform
+    }
+
+    void PythonLoader::UnloadRequiredModules()
+    {
+        // No required modules need explicit unloading on this platform
+    }
+
+    AZ::IO::FixedMaxPath PythonLoader::GetDefault3rdPartyPath(bool createOnDemand)
+    {
+        AZ::IO::FixedMaxPath thirdPartyEnvPathPath;
+
+        // First check if `LY_3RDPARTY_PATH` is explicitly set in the environment
+        size_t envBufferSize{ 0 };
+        getenv_s(&envBufferSize, nullptr, 0, "LY_3RDPARTY_PATH");
+        if (envBufferSize > 0)
+        {
+            AZ_Assert(
+                envBufferSize < AZ::IO::MaxPathLength,
+                "The environment variable for 'LY_3RDPARTY_PATH' must not exceed the max path length of %ld",
+                AZ::IO::MaxPathLength);
+
+            char envBuffer[AZ::IO::MaxPathLength] = {'\0'};
+            getenv_s(&envBufferSize, envBuffer, envBufferSize, "LY_3RDPARTY_PATH");
+            thirdPartyEnvPathPath = AZ::IO::FixedMaxPath(envBuffer);
+        }
+        else
+        {
+            // If not, the generate the default 3rdParty path location which is based on the current user's 
+            // profile path.
+            WCHAR userProfilePath[AZ::IO::MaxPathLength] = { '\0' };
+            HRESULT result = SHGetFolderPathW(NULL, CSIDL_PROFILE | CSIDL_FLAG_CREATE, NULL, 0, userProfilePath);
+            AZ_Assert(SUCCEEDED(result), "Unable to determine profile path needed for the 3rd folder");
+            size_t userProfilePathLength = wcslen(userProfilePath);
+            AZStd::string profilePathUtf8;
+
+            // Convert UTF-16 to UTF-8
+            AZStd::to_string(profilePathUtf8, { userProfilePath, aznumeric_cast<size_t>(userProfilePathLength) });
+
+            AZ::IO::FixedMaxPath profileFixedPath(profilePathUtf8);
```

Comment:
Actually looking at both the Windows and Linux implementations.
There is no need for a platform specific implementation at all.

Using the [AZ::Utils::GetEnv](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L171-L198) function can be used to query environment variables in a portable way across all platforms.
The user's `.o3de` Manifest directory can be queried in a cross-platform way using [AZ::Utils::GetO3deManifestDirectory](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L127-L130) function

Reply:
There is a linux/mac [specific section](https://github.com/spham-amzn/o3de/blob/9759f5d632c83855b947585efd822d5e6f88c332/Code/Framework/AzToolsFramework/Platform/Linux/AzToolsFramework/API/PythonLoader_Linux.cpp#L57-L61) that is not applicable to windows:

Reply:
Why is that not applicable to Windows?

The semantic logic is the same, they are both querying the user's home directory.
On Linux/Mac, it is using `getpwuid` to query the `/etc/passwd` file for the user home directory.
On Windows, it is using `SHGetFolderPathW`

Reply:
Ok, I missed that this was in https://github.com/o3de/o3de/blob/ae19572117eee301495008bebd2d7b97302df840/Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Utils/Utils_UnixLike.cpp#L24 already. Will refactor this.